proto/buf/alpha/registry/v1alpha1/plugin_curation.proto

```diff
@@ -114,6 +116,49 @@ service PluginCurationService {
   rpc GetLatestCuratedPlugin(GetLatestCuratedPluginRequest) returns (GetLatestCuratedPluginResponse);
 }
 
+// PluginCodeGenerationService generates code using remote plugins.
+service PluginCodeGenerationService {
+  // GenerateCode generates code using the specified remote plugins.
+  rpc GenerateCode(GenerateCodeRequest) returns (GenerateCodeResponse);
+}
+
+message GenerateCodeRequest {
+  // The image to run plugins against to generate the desired file outputs.
+  //
+  // All image files that are not imports and not well-known types will be generated.
+  // If you want to filter what files are generated, modify the image.
+  // If you want to include imports, set include_imports.
+  buf.alpha.image.v1.Image image = 1;
+  repeated PluginGenerationRequest requests = 2;
+  // Include imports from the Image in generation.
+  bool include_imports = 3;
+  // Include Well-Known Types from the Image in generation.
+  //
+  // include_imports must be set if include_well_known_types is set.
+  bool include_well_known_types = 4;
+}
+
+message GenerateCodeResponse {
+  // Responses from each plugin execution.
+  // The order of each response matches the order in the GenerateCodeRequest.
+  repeated PluginGenerationResponse responses = 1;
```

Comment:
Wavered a bit on whether to return `CuratedPluginReference` here too but went with simplicity of indexing to find the corresponding response.

Reply:
Sounds good - we'll need to keep this ordered with respect to the requests anyway (re: insertion points), so we can easily start without the `CuratedPluginReference`.